src/discord.ts

```diff
@@ -123,71 +122,94 @@ export const sendLiveAnnouncement = async (streamInfo: StreamInfo) => {
       );
       await message.edit({
         content: `${onlineAnnouncementPrefix}${Discord.Util.escapeMarkdown(
-          streamInfo.user_name
-        )} is now live on Twitch! https://twitch.tv/${streamInfo.user_name}`,
+          info.streamer_info.name
+        )} is now live on ${info.streaming_service.name}! ${info.streaming_service.base_url}${info.streamer_info.name}`,
         embed,
       });
     } else {
       message = await announcementsChannel.send({
         content: `${onlineAnnouncementPrefix}${Discord.Util.escapeMarkdown(
-          streamInfo.user_name
-        )} is now live on Twitch! https://twitch.tv/${streamInfo.user_name}`,
+          info.streamer_info.name
+        )} is now live on ${info.streaming_service.name}! ${info.streaming_service.base_url}${info.streamer_info.name}`,
         embed,
       });
     }
 
     await DiscordAnnouncementModel.updateOne(
-      { memberId: streamInfo.user_id },
+      { memberId: info.streamer_info.id },
       {
-        memberId: streamInfo.user_id,
+        memberId: info.streamer_info.id,
         messageId: message.id,
-        streamId: streamInfo.id,
-        category: category.name,
+        streamId: info.id,
+        category: info.category_name,
       },
       { upsert: true }
     );
   }
 };
-
-export const sendOfflineAnnouncement = async (member_id: string) => {
-  const video = await fetchVideoByUserId(member_id);
-
-  if (!video) {
-    return;
-  }
-
+export const sendOfflineAnnouncement = async (announcement: StreamAnnouncementInfo) => {
   const saved_message = await DiscordAnnouncementModel.findOne({
-    memberId: member_id,
+    memberId: announcement.streamer_info.id,
   });
 
   if (!saved_message) {
     return;
   }
-
   const message = await announcementsChannel.messages.fetch(
     `${saved_message.messageId}`
   );
-
-  const user = await UserManager.getUserById(member_id);
-
+  let footer = "Finished streaming";
+  if (announcement.archivedStream) {
+    footer += ` • Streamed for ${announcement.archivedStream.duration}`;
```

Comment:
I understand that this isn't available from the Glimesh API. Can we calculate it somehow? From the age of the original message perhaps?